File: documentation-examples-kotlin/src/kotlin/dev/typr/foundationskt/docs/core/QueryListenerBasic.kt
package dev.typr.foundationskt.docs.core

import dev.typr.foundationskt.*
import dev.typr.foundationskt.data.*

@Suppress("unused")
class QueryListenerBasic {
    //start
    object logger : QueryListener {
        override fun beforeQuery(sql: String, name: java.util.Optional<String>) {
            println("Executing: $sql")
        }
        override fun afterQuery(event: QueryEvent) {
            println("${event.name().orElse("unnamed")} completed in ${event.elapsed().toMillis()}ms")
        }
        override fun failedQuery(event: QueryEvent) {
            System.err.println("${event.name().orElse("unnamed")} failed after ${event.elapsed().toMillis()}ms: ${event.error().map { it.message }.orElse("unknown")}")
        }
    }
    //stop
}

File: documentation-examples-scala/src/scala/dev/typr/foundationssc/docs/core/QueryListenerBasic.scala
package dev.typr.foundationssc.docs.core
import dev.typr.foundationssc.*
import dev.typr.foundationssc.data.*

import dev.typr.foundations.{QueryListener, QueryEvent}

@SuppressWarnings(Array("unused"))
object QueryListenerBasic:
  // start
  object logger extends QueryListener:
    override def beforeQuery(sql: String, name: java.util.Optional[String]): Unit =
      println(s"Executing: $sql")
    override def afterQuery(event: QueryEvent): Unit =
      println(s"${event.name().orElse("unnamed")} completed in ${event.elapsed().toMillis}ms")
    override def failedQuery(event: QueryEvent): Unit =
      System.err.println(s"${event.name().orElse("unnamed")} failed after ${event.elapsed().toMillis}ms: ${event.error().map(_.getMessage).orElse("unknown")}")
  // stop
